contracts/ERC721TokenRegistry.sol

```diff
@@ -0,0 +1,193 @@
+pragma solidity 0.4.18;
+
+import "zeppelin-solidity/contracts/ownership/Ownable.sol";
+
+
+/**
+ * The ERC721TokenRegistry is a basic registry mapping token symbols
+ * to their known, deployed addresses on the current blockchain.
+ *
+ * Note that the TokenRegistry does *not* mediate any of the
+ * core protocol's business logic, but, rather, is a helpful
+ * utility for Terms Contracts to use in encoding, decoding, and
+ * resolving the addresses of currently deployed tokens.
+ *
+ * At this point in time, administration of the Token Registry is
+ * under Dharma Labs' control.  With more sophisticated decentralized
+ * governance mechanisms, we intend to shift ownership of this utility
+ * contract to the Dharma community.
+ */
+contract ERC721TokenRegistry is Ownable {
+    mapping (bytes32 => TokenAttributes) public symbolHashToTokenAttributes;
+    string[4294967296] public tokenSymbolList;
+    uint32 public tokenSymbolListLength;
+
+    struct TokenAttributes {
+        // The ERC721 contract address.
+        address tokenAddress;
+        // The index in `tokenSymbolList` where the token's symbol can be found.
+        uint tokenIndex;
+        // The name of the given token, e.g. "CryptoKitties"
+        string name;
+    }
+
+    /**
+     * Maps the given symbol to the given token attributes.
+     */
+    function setTokenAttributes(
+        string _symbol,
+        address _tokenAddress,
+        string _tokenName
+    )
+        public onlyOwner
+    {
+        bytes32 symbolHash = keccak256(_symbol);
+
+        // Attempt to retrieve the token's attributes from the registry.
+        TokenAttributes memory attributes = symbolHashToTokenAttributes[symbolHash];
+
+        if (attributes.tokenAddress == address(0)) {
+            // The token has not yet been added to the registry.
+            attributes.tokenAddress = _tokenAddress;
+            attributes.name = _tokenName;
+            attributes.tokenIndex = tokenSymbolListLength;
+
+            tokenSymbolList[tokenSymbolListLength] = _symbol;
+            tokenSymbolListLength++;
+        } else {
+            // The token has already been added to the registry; update attributes.
+            attributes.tokenAddress = _tokenAddress;
+            attributes.name = _tokenName;
+        }
+
+        // Update this contract's storage.
+        symbolHashToTokenAttributes[symbolHash] = attributes;
+    }
+
+    /**
+     * Given a symbol, resolves the current address of the token the symbol is mapped to.
+     */
+    function getTokenAddressBySymbol(string _symbol) public view returns (address) {
+        bytes32 symbolHash = keccak256(_symbol);
+
+        TokenAttributes storage attributes = symbolHashToTokenAttributes[symbolHash];
```

Comment:
I believe this can be referred to as memory, not storage, given that you're not writing to disk. i believe this also saves gas.

Reply:
I don't believe that this uses any gas, see:

> When a storage variable is created in function locally by look up, it simply reference data already allocated on Storage. No new storage is created.
- https://medium.com/coinmonks/ethereum-solidity-memory-vs-storage-which-to-use-in-local-functions-72b593c3703a